admin/spam/views.py

```diff
@@ -1,96 +1,124 @@
-        spam_id = kwargs.get('spam_id', None)
-        self.spam_id = spam_id
-        self.page = request.GET.get('page', 1)
         try:
-            self.spam = serialize_comment(Comment.load(spam_id))
-        except (AttributeError, TypeError):
-            return HttpResponseNotFound(
-                '<h1>Spam comment ({}) not found.</h1>'.format(spam_id)
+            return super(EmailFormView, self).get(request, *args, **kwargs)
+        except AttributeError:
+            return page_not_found(
+                request,
+                AttributeError(
+                    'Spam with id {} not found.'.format(kwargs.get('spam_id'))
+                )
             )
-        form = self.get_form()
-        context = {
-            'comment': self.spam,
-            'page_number': request.GET.get('page', 1),
-            'form': form
-        }
-        return render(request, self.template_name, context)
 
     def post(self, request, *args, **kwargs):
-        spam_id = kwargs.get('spam_id', None)
-        self.spam_id = spam_id
-        self.page = request.GET.get('page', 1)
         try:
-            self.spam = serialize_comment(Comment.load(spam_id))
-        except (AttributeError, TypeError):
-            return HttpResponseNotFound(
-                '<h1>Spam comment ({}) not found.</h1>'.format(spam_id)
+            return super(EmailFormView, self).post(request, *args, **kwargs)
+        except AttributeError:
+            return page_not_found(
+                request,
+                AttributeError(
+                    'Spam with id {} not found.'.format(kwargs.get('spam_id'))
+                )
             )
-        return super(EmailFormView, self).post(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        kwargs.setdefault(
+            'comment',
+            serialize_comment(Comment.load(self.kwargs.get('spam_id')))
+        )
+        kwargs.setdefault('page_number', self.request.GET.get('page', '1'))
+        kwargs.setdefault('status', self.request.GET.get('status', '1'))
+        return super(EmailFormView, self).get_context_data(**kwargs)
 
     def get_initial(self):
+        spam = serialize_comment(Comment.load(self.kwargs.get('spam_id')))
         self.initial = {
-            'author': self.spam['author'].fullname,
-            'email': [(r, r) for r in self.spam['author'].emails],
+            'author': spam['author'].fullname,
+            'email': [(r, r) for r in spam['author'].emails],
             'subject': 'Reports of spam',
             'message': render(
                 None,
                 'spam/email_template.html',
-                {'item': self.spam}
+                {'item': spam}
             ).content,
         }
         return super(EmailFormView, self).get_initial()
 
     def form_valid(self, form):
+        message = form.cleaned_data.get('message')
+        email = form.cleaned_data.get('email')
+        update_admin_log(
```

Comment:
Would you maybe want to log after you email? If there's an exception sending the mail we will have a log of something that didn't happen.